### PR TITLE
TS Strict Mode

### PR DESCRIPTION
**Pull request by amirafzali**
Updated majority of /admin to be compliant with TS strict mode. Improved many typings.

### CODE REVIEW

File: admin/src/components/ApplicantDropdown.vue [hunk 1]
```diff
@@ -245,13 +246,15 @@ export default {
   components: {
     vueSlider,
   },
-  computed: {
+  /*
+  created: {
```
Comment (kumailn) [inline admin/src/components/ApplicantDropdown.vue]: Why were the computed properties removed?

File: admin/src/components/Navbar.vue [hunk 2]
```diff
@@ -41,16 +41,17 @@
...
+export default Vue.extend({
+  name: 'Navbar',
   components: {},
   data() {
     return {
-      current: "DH V",
+      current: 'DH V',
       c_user: firebase.auth().currentUser,
-      dhs: ["DH V", "DH IV", "DH III", "DH II"]
+      dhs: ['DH V', 'DH IV', 'DH III', 'DH II'],
```
Comment (kumailn) [inline admin/src/components/Navbar.vue]: ty for linting 🎉

File: admin/src/components/Navbar.vue [hunk 3]
```diff
@@ -60,15 +61,15 @@ export default {
         .signOut()
         .then(
           () => {
-            this.$router.push({ name: "Login" });
+            this.$router.push({ name: 'Login' });
           },
-          error => {
+          (error) => {
```
Comment (kumailn) [inline admin/src/components/Navbar.vue]: can you ensure your linter doesn't parenthesize single params, we're following the Airbnb styleguide. https://github.com/airbnb/javascript#arrows--one-arg-parens

File: admin/src/views/AdminSignup.vue [hunk 4]
```diff
@@ -147,7 +171,7 @@ export default {
           if (adminSignupResponse.data.createdUser) {
             this.$router.push({
               name: 'Login',
-              params: { successFeedback: true },
+              params: { successFeedback: 'true' },
```
Comment (kumailn) [inline admin/src/views/AdminSignup.vue]: hmm, why was this changed to a string? line 117 references it as a boolean 🤔

Comment (amirafzali) [reply admin/src/views/AdminSignup.vue]: True, will change that
